src/test/java/com/github/checkstyle/regression/git/DiffParserTest.java

```diff
@@ -98,4 +159,48 @@ public void testParsePrBranchBehindMaster() throws Exception {
                     expected, changes.get(0));
         }
     }
+
+    @Test
+    public void testParsePrBranchWithMultipleCommits() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            GitUtils.addAnEmptyFileAndCommit(repository, "AddInCommit1");
+            GitUtils.addAnEmptyFileAndCommit(repository, "AddInCommit2");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals("There should be 2 change detected", 2, changes.size());
+            final GitChange expected0 = ImmutableGitChange.builder()
+                    .path("AddInCommit1")
+                    .build();
+            assertEquals("The change is not as expected", expected0, changes.get(0));
+            final GitChange expected1 = ImmutableGitChange.builder()
+                    .path("AddInCommit2")
+                    .build();
+            assertEquals("The change is not as expected", expected1, changes.get(1));
+        }
+    }
+
+    @Test
+    public void testParseFilePermissionChange() throws Exception {
+        // Skip this UT on Windows
+        if (!SystemUtils.IS_OS_WINDOWS) {
+            try (Repository repository = GitUtils.createNewRepository()) {
+                final File file = GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+                GitUtils.createNewBranchAndCheckout(repository, "foo");
+                final Set<PosixFilePermission> permissions =
+                        Files.getPosixFilePermissions(file.toPath());
+                permissions.add(PosixFilePermission.OWNER_EXECUTE);
+                Files.setPosixFilePermissions(file.toPath(), permissions);
+                GitUtils.addAllAndCommit(repository, "Change file mode");
+                final List<GitChange> changes = DiffParser.parse(
+                        repository.getDirectory().getParent(), "foo");
+                assertEquals("There should be 1 change detected", 1, changes.size());
```

Comment:
If the only change is a chmod (no changes to the contents of the file), I think we should ignore the change altogether.
Chmod has no bearing on the final contents of the JAR, which doesn't support chmod.
If you agree, make a new issue for it. We will require a new test in that issue, chmod plus file change in 1 commit.